### ISSUE TITLE
Large regression average_frame_rasterizer_time_millisworst_frame_rasterizer_time_millis new_gallery__transition_perf)

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#66989]
https://flutter-flutter-perf.skia.org/e/?begin=1601326624&end=1601400498&keys=X5ceea3c4ffa55faa8a466320f7a3c13d&num_commits=50&request_type=1&xbaroffset=21690.

Looks like a gallery roll? https://github.com/flutter/flutter/pull/66693.

cc @liyuqian, the committer.

Leaving high-priority to investigate, even if the issue is expected, until we know more.

**Comment by liyuqian** [comment]
@willlarche @guidezpl : we've seen a 1.9x worst frame speed regression (290ms to 550ms) after a gallery roll #66693. Can you please prioritize how severe this is for the published gallery app, and whether you'd like to spend some time to bisect your commit figure out if you want to revert some gallery commits?

**Comment by guidezpl** [comment]
I had a go at bisecting 
* Flutter (Channel master, 1.22.0-10.0.pre.433, on Mac OS X 10.15.6 19G2021 x86_64,
    locale en-US) 
* Pixel 3a (android-arm64, Android 10)
* `flutter drive --profile --trace-startup -t test_driver/transitions_perf.dart`

The results were inconclusive. I saw a `worst_frame_rasterizer_time_millis` of `132ms` for the gallery at `a208eac6e6e8336ae9820e54c572c099231f1da2`, and `140ms` at `53462151bf753dc70284be9274a798fa485265f2 ` with values oscillating between `136ms` and `150ms` in between.

Do you have ideas to try and understand the benchmark results?

**Comment by liyuqian** [comment]
Maybe you need a Moto G4 to reproduce this regression. If @kf6gpe has one, he can mail it to you if that's convenient. Otherwise, Eric suggested to directly gCard one from Amazon.

**Comment by kf6gpe** [comment]
I do not have a G4 in inventory, I'm afraid.

**Comment by guidezpl** [comment]
I'll get a G4 to investigate

**Comment by chinmaygarde** [comment]
Triage: Jason thinks the application was updated around the same time as the regression. Waiting @guidezpl to get back to us on the reproduction.

**Comment by kf6gpe** [comment]
@guidezpl How's it going with the bisect? Need anything from our side of the world?

**Comment by guidezpl** [comment]
Hey Ray, still waiting on the device, slated to arrive tomorrow or Thursday. I'll post an update as soon as I can do a bisect

**Comment by kf6gpe** [comment]
(Weekly performance triage) Any luck, @guidezpl? Let me know if I should stop joggling your elbow for a bit, too. Thanks!

**Comment by guidezpl** [comment]
Got it yesterday, currently bisecting :)

**Comment by liyuqian** [comment]
Option 1 sounds good for now so we can close this issue. (No actual rebaseline is needed as Skia perf doesn't need a baseline and it uses this issue to track if the change is accepted or not.)

In the future, maybe the gallery app can write perf tests that are more focused (option 2) so new pages won't regress old tests.

Meanwhile, 500ms worst frame time may look very bad to a user. Have you considered to use https://flutter.dev/docs/perf/rendering/shader to reduce that worst frame time?

**Comment by guidezpl** [comment]
Sounds good. Definitely considered it. I've filed https://github.com/flutter/gallery/issues/354 to track future work
